nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/services/azure/cosmos/document/AzureCosmosSecureDBClientService.java

```diff
@@ -0,0 +1,158 @@
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.processors.azure.cosmos.document.AzureCosmosDBUtils;
+import org.apache.nifi.services.azure.cosmos.AzureCosmosDBConnectionService;
+import org.apache.nifi.services.azure.keyvault.AzureKeyVaultConnectionService;
+import org.apache.nifi.util.StringUtils;
+
+@Tags({"azure", "cosmos", "document", "service", "secure"})
+@CapabilityDescription(
+        "Provides a controller service that configures a connection to Cosmos DB (Core SQL API) " +
+                " and provides access to that connection to other Cosmos DB-related components."
+)
+public class AzureCosmosSecureDBClientService
+        extends AbstractCosmosDBClientService
+        implements AzureCosmosDBConnectionService {
+
+    private String uriSecret;
+    private String accessKeySecret;
+    private String consistencyLevel;
+    private AzureKeyVaultConnectionService keyVaultClientService;
+
+    public static final PropertyDescriptor URI_SECRET = new PropertyDescriptor.Builder()
+            .name("azure-cosmos-db-uri-secret")
+            .displayName("Cosmos DB URI Secret Name")
+            .description("Cosmos DB URI Secret Name")
+            .required(true)
+            .addValidator(StandardValidators.URI_VALIDATOR)
+            .sensitive(true)
+            .build();
+
+    public static final PropertyDescriptor DB_ACCESS_KEY_SECRET = new PropertyDescriptor.Builder()
+            .name("azure-cosmos-db-key-secret")
+            .displayName("Cosmos DB Access Key Secret Name")
+            .description("Cosmos DB Access Key Secret Name")
+            .required(true)
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .sensitive(true)
+            .build();
+
+    static final PropertyDescriptor KEYVAULT_CONNECTION_SERVICE = new PropertyDescriptor.Builder()
+            .name("azure-keyvault-connection-service")
+            .displayName("KeyVault Connection Service")
+            .description("If configured, the controller service used to obtain the secrets from KeyVault")
+            .required(true)
+            .identifiesControllerService(AzureKeyVaultConnectionService.class)
+            .build();
+    @OnEnabled
+    public void onEnabled(final ConfigurationContext context) {
+        this.uriSecret = context.getProperty(URI_SECRET).getValue();
+        this.accessKeySecret = context.getProperty(DB_ACCESS_KEY_SECRET).getValue();
+        this.consistencyLevel = context.getProperty(
+                AzureCosmosDBUtils.CONSISTENCY).getValue();
+        this.keyVaultClientService = context.getProperty(
+                KEYVAULT_CONNECTION_SERVICE
+        ).asControllerService(AzureKeyVaultConnectionService.class);
+
+        createCosmosClient(
+                getURI(),
+                getAccessKey(),
+                getConsistencyLevel()
+        );
+    }
+
+    static List<PropertyDescriptor> descriptors = new ArrayList<>();
```

Comment:
Following standard conventions, static variables and initializers should be declared before non-static methods.

Reply:
fixed